api-gateway/src/main/resources/application.yml

```diff
@@ -17,26 +17,26 @@ spring:
     gateway:
       routes:
         - id: api_route
-          uri: http://backend:5000
+          uri: http://backend:5800
```

Comment:
Please check if spring boot configuration supports referencing properties from other properties, so it could be rewritten as `uri: ${gateway.backend.url}`

Reply:
Yep, seems like to works.